kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSFunctionDeclarationImpl.kt

```diff
@@ -59,7 +66,19 @@ class KSFunctionDeclarationImpl private constructor(internal val ktFunctionSymbo
         }
     }
 
+    @OptIn(SymbolInternals::class)
     override val returnType: KSTypeReference? by lazy {
+        // FIXME: temporary workaround before upstream fixes java type refs.
+        if (origin == Origin.JAVA) {
+            if (ktFunctionSymbol is KtFirFunctionSymbol) {
+                (ktFunctionSymbol.firSymbol.fir as? FirJavaMethod)?.also {
```

Comment:
Can you replace the type checks with `KtFirSymbol` and `FirCallableDeclaration`, hoist the workaround and share among the 3 places?

If possible, please keep those in a `WorkaroundsUsingInternals.kt` so that we won't forget to clean them up before formal release.

Reply:
The reason I put this at different places is that, unfortunately `FirCallableDeclaration.returnTypeRef` is declared as a `val`, and it only gets overridden to `var` in `FirJavaMethod` , `FirJavaField` and `FirJavaValueParameter` (somehow makes total sense since they are the ones need to be transformed)